src/libstd/sys/unix/fs.rs

```diff
@@ -108,6 +108,27 @@ impl FilePermissions {
     pub fn mode(&self) -> raw::mode_t { self.mode }
 }
 
+#[unstable(feature = "file_types_posix", reason = "recently added API")]
+pub trait FileTypeUnix {
```

Comment:
Conventionally we'd call this trait `FileTypeExt` and this would be placed in the `std::os::unix::fs` module, located at `src/libstd/sys/unix/ext/fs.rs`